snakemake-mode.el

```diff
@@ -325,6 +328,57 @@ label."
     (nreverse index)))
 
 
+;;; Embedded language syntax-highlighting
+
+(defun snakemake-mode-setup-mmm ()
+  "Set up MMM mode to highlight embedded R code.
+
+You must have the R-strings either within a R(''' ''') function call or a code
+block delimited with '''#r and ''' (triple double-quotes also accepted).
+
+For automatic highlighting of embedded regions, you need to set mmm-global-mode
+to 'maybe in your setup file."
+
+  (progn
+    (unless (featurep 'mmm-mode)
+      (error "You need to install mmm-mode"))
+
```

Comment:
s/error/user-error/

`featurep` is wrong here because it only tests if something is loaded
(sorry, I know I suggested it).  Instead you can go back to `require`
with `NOERROR` as non-nil.

    (unless (require 'mmm-mode nil t)
      (user-error ...))